tests/unit_tests/test_build_config.py

```diff
@@ -0,0 +1,36 @@
+# ##############################################################################
+#  (c) Crown copyright Met Office. All rights reserved.
+#  For further details please refer to the file COPYRIGHT
+#  which you should have received as part of this distribution
+# ##############################################################################
+from textwrap import dedent
+from unittest import mock
+
+from fab.build_config import BuildConfig
+from fab.dep_tree import AnalysedFile
+from fab.steps.compile_fortran import CompileFortran
+
+
+class TestBuildConfig(object):
+
+    def test_error_newlines(self, tmp_path):
```

Comment:
Is the purpose of this test to ensure that newlines are preserved through the error reporting process. It's not clear to me how much of what is being tested.

Reply:
Yes that's right. When there's an error containing new lines, v0.9.0a1 does not display the newlines, it prints `\\n`. I've tried to make that more clear in the test's comment.